pkg/version/fixbuild.go

```diff
@@ -2,21 +2,48 @@
 
 package version
 
-import "runtime/debug"
+import (
+	"os/exec"
+	"runtime/debug"
+	"strings"
+)
 
 func init() {
 	fixBuild = buildInfoFixBuild
 }
 
+func getGitCommitHash() string {
+	cmd := exec.Command("git", "rev-parse", "HEAD")
+	cmd.Dir = "."
+
+	output, err := cmd.Output()
+	if err != nil {
+		return ""
+	}
+
+	return strings.TrimSpace(string(output))
+}
+
 func buildInfoFixBuild(v *Version) {
-	info, ok := debug.ReadBuildInfo()
-	if !ok {
+	// First, try to get the hash from ldflags
+	if v.Build != "$Id$" && v.Build != "" {
 		return
 	}
-	for i := range info.Settings {
-		if info.Settings[i].Key == "gitrevision" {
-			v.Build = info.Settings[i].Value
-			break
+
+	// Try build info
+	info, ok := debug.ReadBuildInfo()
+	if ok {
+		for _, setting := range info.Settings {
+			if setting.Key == "vcs.revision" {
```

Comment:
At some point the key in settings that we wanted was gitrevision, I don't know when it was changed but I doesn't hurt to still check for it.

Reply:
Thanks for reviewing quickly. I'll look for it and resolve other comments